lib/fog/openstack/identity.rb

```diff
@@ -14,16 +14,21 @@ class OpenStack < Fog::Service
                  :current_user, :current_user_id, :current_tenant, :openstack_cache_ttl,
                  :provider, :openstack_identity_prefix, :openstack_endpoint_path_matches
 
-      # Fog::Identity::OpenStack.new() will return a Fog::Identity::OpenStack::V2 or a Fog::Identity::OpenStack::V3,
-      #  depending on whether the auth URL is for an OpenStack Identity V2 or V3 API endpoint
+      # Fog::Identity::OpenStack.new() will return a Fog::Identity::OpenStack::V3 by default
       def self.new(args = {})
-        if self.inspect == 'Fog::Identity::OpenStack'
-          identity = super
-          config = identity.config
-          service = identity.v3? ? Fog::Identity::OpenStack::V3.new(config) : Fog::Identity::OpenStack::V2.new(config)
-        else
-          service = Fog::Service.new(args)
+        version = '3'
+        url = Fog.credentials[:openstack_auth_url] || args[:openstack_auth_url]
```

Comment:
@gildub @dhague doh!, Fog.credentials[:openstack_auth_url]  should not be really here?

That means it always takes version of the API from my ~/.fog but then other params from the request.